### PR TITLE
 Add optional type annotations to use assignment 

### PR DESCRIPTION
**Pull request by marcinkoziej**
In reference to #56 

I think I've added the optional type annotation correctly.
Test passes.
I am not sure if queries need to be updated somewhere?

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -424,7 +424,11 @@ module.exports = grammar({
     use: ($) =>
       seq(
         "use",
-        optional(series_of($._pattern, ",")),
+        optional(series_of(
+          seq(
+            $._pattern,
+            optional($._type_annotation)
+          ), ",")),
```
Comment (the-mikedavis) [inline grammar.js:431]: Looks like this needs a run of the formatter. If you use `npm format` it should run prettier and fix the CI

File: src/parser.c
Comment (the-mikedavis) [inline src/parser.c:1]: No need to commit the files in `src/*`: these are automatically generated and committed by the CI

File: grammar.js [hunk 2]
```diff
@@ -453,7 +453,7 @@ module.exports = grammar({
     use: ($) =>
       seq(
         "use",
-        optional(series_of($._pattern, ",")),
+        optional(series_of(seq($._pattern, optional($._type_annotation)), ",")),
```
Comment (J3RN) [inline grammar.js:456]: I think we should make a `use_assignments` node and a `use_assignment` node (corresponding to the `UseAssignment` node in the Gleam parser) here.  My rationale is just that the current structure is a bit difficult for humans (and likely machines) to understand.  For instance, the following code is valid:

```gleam
use a, b: String, c: Int, d <- foo()
```

The AST generated by `tree-sitter-gleam would look like this:
```scheme
(use
  (identifier)
  (identifier)
  (type
    (type_identifier))
  (identifier)
  (type
    (type_identifier))
  (identifier)
  (function_call
    (identifier)
    (arguments)))
```

To me, it's a bit hard to see what's going on.  The alternative would be something like this:
```suggestion
        optional(field("assignments", $.use_assignments), ",")),
```

yeilding this:

```scheme
(use
  (use_assignments
    (use_assignment
      (identifier))
    (use_assignment
      (identifier)
      (type
        (type_identifier)))
    (use_assignment
      (identifier)
      (type
        (type_identifier)))
    (use_assignment
      (identifier)))
  (function_call
    (identifier)
    (arguments)))
```

Notably the Gleam parser has no `UseAssignments` node, but then again the Gleam parser has no `FunctionParameters` node either but we have a `function_parameters` node; we have to compensate for not really having "lists".

**Review by J3RN — COMMENTED** [review]
Hey @marcinkoziej, thank you for your PR! :sparkles:  I'm sorry it's been almost two months :sweat:  I rebased your branch to compensate for some vigorous rebasing upstream and ran the formatter as @the-mikedavis suggested.

I have just one additional ask for this (commented inline).

File: grammar.js (hunk 2 above)
Comment (marcinkoziej) [reply grammar.js:456]: Hi! Thanks for reflecting on this!
I am very n00b about how to do tree-sitter grammars and I suppose what you propose is absolutely a good idea!

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**